### PR TITLE
ci: Run style transform in Storybook

### PR DESCRIPTION
**Pull request by NicholasBoll**
## Summary

Fixes: #2630 

Runs the Style transform as part of Webpack so we can test the style transform both visually in Chromatic and in development. We'll also get any errors caused by incorrect styling.

- The style transform in Webpack shares the same TypeScript program as the doc parser. The Webpack processing time is about the same as just running the Doc Parser (doc parser is slow with all those type checker calls)
- Style transform is on by default in Storybook in development. We can see errors the style transform will give during a CI build in Webpack
- Chromatic will run visual regression against our style transform! So we can see any style transform issues BEFORE our users see them.
- `SKIP_DOCGEN=true` controls both the processing by the docgen as well as the style transform. Disabling these preprocessors is a much faster experience for both getting Storybook started as well as hot reloading. But docgen/style transform should be run for accuracy at least before pushing changes to a PR

## Release Category
Infrastructure

---

## Checklist

- [ ] MDX documentation adheres to Canvas Kit's [Documentation Guidelines](https://workday.github.io/canvas-kit/?path=/docs/guides-documentation-guidelines--page)
- [x] Label `ready for review` has been added to PR

## For the Reviewer

- [x] PR title is short and descriptive
- [x] PR summary describes the change (Fixes/Resolves linked correctly)

## Where Should the Reviewer Start?

Test files

## Areas for Feedback? (optional)

- [ ] Code
- [ ] Testing

## Testing Manually

Run storybook locally using `yarn start`. You cannot skip docs.

### CODE REVIEW

File: modules/docs/docgen/createDocProgram.js [hunk 1]
```diff
@@ -34,8 +34,10 @@ function getTSConfig(basePath = '.') {
     const contents = ts.readConfigFile(tsconfigPath, ts.sys.readFile);
     if (contents.config) {
       // https://github.com/microsoft/TypeScript/issues/5276#issuecomment-149369652
-      config = ts.convertCompilerOptionsFromJson(contents.config.compilerOptions, tsconfigPath)
-        .options;
+      config = ts.convertCompilerOptionsFromJson(
+        contents.config.compilerOptions,
+        tsconfigPath
+      ).options;
```
Comment (NicholasBoll) [inline modules/docs/docgen/createDocProgram.js:40]: This is a prettier change since we upgraded to Prettier v2. First change to this file since then.

File: .storybook/main.js [hunk 2]
```diff
@@ -64,8 +67,18 @@ module.exports = {
         include: [modulesPath],
         exclude: /examples|stories|spec|codemod|docs/,
         loaders: [
+          // loaders are run in reverse order. symbol-doc-loader needs to be done first
+          {
+            loader: path.resolve(__dirname, 'style-transform-loader'),
+            options: {
+              Doc,
+            },
+          },
```
Comment (mannycarrera4) [inline .storybook/main.js:76]: is this what adds the transform as part of webpack?

File: .storybook/symbol-doc-loader.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 // @ts-check
-const {createDocProgram} = require('../modules/docs/docgen/createDocProgram');
```
Comment (mannycarrera4) [inline .storybook/symbol-doc-loader.js:2]: why was this removed?

Comment (josh-bagwell) [reply .storybook/symbol-doc-loader.js:2]: Just making a note here, I think it's because Nicholas is creating the program now in `main.js`. Could be wrong though

Comment (NicholasBoll) [reply .storybook/symbol-doc-loader.js:2]: Yes. The program is created in `main.js`. I think eventually it makes sense to have a Webpack plugin that manages the program instead though.

File: .storybook/main.js (hunk 2 above)
Comment (NicholasBoll) [reply .storybook/main.js:76]: Yes
